Source/ChocolateyGui/App.xaml.cs

```diff
@@ -32,30 +32,49 @@ public App()
             {
                 var requestedAssembly = new AssemblyName(args.Name);
 
+                try
+                {
+                    if (string.Equals(requestedAssembly.Name, "chocolatey", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var installDir = Environment.GetEnvironmentVariable("ChocolateyInstall");
+                        if (string.IsNullOrEmpty(installDir))
+                        {
+                            var rootDrive = Path.GetPathRoot(Assembly.GetExecutingAssembly().Location);
+                            if (string.IsNullOrEmpty(rootDrive))
+                            {
+                                return null; // TODO: Maybe return the chocolatey.dll file instead?
+                            }
+
+                            installDir = Path.Combine(rootDrive, "ProgramData", "chocolatey");
+                        }
+
+                        var assemblyLocation = Path.Combine(installDir, "choco.exe");
+
+                        return AssemblyResolver.ResolveOrLoadAssembly("choco", string.Empty, assemblyLocation);
+                    }
+
 #if FORCE_CHOCOLATEY_OFFICIAL_KEY
                 var chocolateyGuiPublicKey = Bootstrapper.OfficialChocolateyPublicKey;
 #else
-                var chocolateyGuiPublicKey = Bootstrapper.UnofficialChocolateyPublicKey;
+                    var chocolateyGuiPublicKey = Bootstrapper.UnofficialChocolateyPublicKey;
 #endif
 
-                try
-                {
-                    if (requestedAssembly.get_public_key_token().is_equal_to(chocolateyGuiPublicKey)
-                        && requestedAssembly.Name.is_equal_to(Bootstrapper.ChocolateyGuiCommonAssemblySimpleName))
+                    if (AssemblyResolver.IsPublicKeyToken(requestedAssembly, chocolateyGuiPublicKey)

```

Comment:
`IsPublicKeyToken` not sure about the name here. You are comparing one public key token to another, right? Should this be something like `DoesPublicKeyTokenMatch`? Naming is hard 😢